root/static/scripts/common/components/ArtistCreditLink.js

```diff
@@ -1,38 +1,53 @@
 /*
+ * @flow
  * This file is part of MusicBrainz, the open internet music database.
  * Copyright (C) 2015–2016 MetaBrainz Foundation
  * Licensed under the GPL version 2, or (at your option) any later version:
  * http://www.gnu.org/licenses/gpl-2.0.txt
  */
 
-const React = require('react');
+import React from 'react';
 
-const EntityLink = require('./EntityLink');
+import EntityLink, {DeletedLink} from './EntityLink';
 
-const ArtistCreditLink = ({artistCredit, showDeleted = true, ...props}) => {
+type Props = {
+  +artistCredit: ArtistCreditT,
+  +plain?: boolean,
+  +showDeleted?: boolean,
+  +target?: '_blank',
+};
+
+const ArtistCreditLink = ({
+  artistCredit,
+  showDeleted = true,
+  ...props
+}: Props) => {
   const parts = [];
   for (let i = 0; i < artistCredit.length; i++) {
     const credit = artistCredit[i];
     if (props.plain) {
       parts.push(credit.name);
     } else {
-      let artist = credit.artist;
-      if (!artist) {
-        artist = {entityType: 'artist', name: credit.name};
+      const artist = credit.artist;
+      if (artist) {
+        parts.push(
+          <EntityLink
+            content={credit.name}
+            entity={artist}
+            key={i}
+            showDeleted={showDeleted}
+            target={props.target}
+          />,
+        );
+      } else {
+        parts.push(
+          <DeletedLink allowNew={false} name={credit.name} />,
```

Comment:
Instead of sending a "fake artist" to EntityLink (which was a flow mess), and since we just get a DeletedLink in this case anyway (because gid was missing), we might as well just export DeletedLink and use it here.